### PR TITLE
Normalize input difference

### PR DESCRIPTION
**Pull request by adumasphi**
Coming back to this [issue](http://trac.openturns.org/ticket/859), the problem was coming from the different order of magnitude of the input variable. In the example of the issue, one variable is around 30000 whereas all the others are small. When you compute the finite difference, we must normalize the difference.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ install:
   - sudo apt-get update
   - sudo apt-get -y install libopenturns-dev python-openturns cmake gfortran swig texlive-latex-recommended texlive-fonts-recommended texlive-latex-extra python-matplotlib
   - sudo rm -r /opt/python/*
-  - pip install sphinx numpydoc matplotlib --user
+  - pip install sphinx numpydoc matplotlib --user --upgrade certifi
```
Comment (jschueller) [inline .travis.yml]: I found out that adding "--upgrade" was enough, no need to specify the "certifi" module

**Review by sofianehaddad — APPROVED** [review]

**Review by sofianehaddad — COMMENTED** [review]
The normalization should be done using the "true" bounds and not sample ones (see https://github.com/cran/sensitivity/blob/master/R/morris.R#L131 for example)

Thus maybe it requires updating the API ?

**Comment by adumasphi** [comment]
Yes to be sure we have the proper bounds, we must give it to the analysis.
I propose : 
- to add a parameter "interval" in the analysis for the constructor Morris(inputSample, outputSample)
- to add a method to MorrisExperiment that  returns the interval in order to get it in the analysis.

File: lib/src/otmorris/MorrisExperiment.hxx [hunk 2]
```diff
@@ -58,6 +58,9 @@ public:
   /** Virtual constructor method */
   MorrisExperiment * clone() const;
 
+  /* Get the interval values */
+  OT::Interval getInterval() const;
```
Comment (jschueller) [inline lib/src/otmorris/MorrisExperiment.hxx]: I'd rename it getBounds for consistency with what's already in OT

File: lib/src/Morris.cxx [hunk 3]
```diff
@@ -92,6 +94,8 @@ Morris::Morris(const MorrisExperiment & experiment, const Function & model)
   if (size != N * (inputDimension + 1))
     throw InvalidArgumentException(HERE) << "In Morris::Morris, sample size should be a multiple of " << inputDimension + 1;
 
+  // get the bounds
+  interval_ = experiment.getInterval();
```
Comment (jschueller) [inline lib/src/Morris.cxx]: you can assign this in the initializer list

**Review by jschueller — APPROVED** [review]

File: lib/src/Morris.cxx [hunk 4]
```diff
@@ -119,7 +122,7 @@ void Morris::computeEffects(const UnsignedInteger N)
     {
       // Evaluate dx
       for (UnsignedInteger j = 0; j < inputDimension; ++j)
-        dx(i, j) = inputSample_[blockIndex + i + 1][j] - inputSample_[blockIndex + i][j];
+        dx(i, j) = (inputSample_[blockIndex + i + 1][j] - inputSample_[blockIndex + i][j]) / diff_bounds[j];
       // Evaluate dy
```
Comment (sofianehaddad) [inline lib/src/Morris.cxx]: dx(i, j) = (inputSample_(blockIndex + i + 1, j) - inputSample_(blockIndex + i, j)) / diff_bounds[j];

File: lib/src/otmorris/Morris.hxx [hunk 5]
```diff
@@ -41,7 +41,7 @@ public:
   Morris();
 
   /** Standard constructor with in/out designs */
-  Morris(const OT::Sample & inputSample, const OT::Sample & outputSample);
+  Morris(const OT::Sample & inputSample, const OT::Sample & outputSample, const OT::Interval & interval);
```
Comment (sofianehaddad) [inline lib/src/otmorris/Morris.hxx:45]: I think we could also get Morris(const OT::Sample & inputSample, const OT::Sample & outputSample);
In that case, we consider interval_ = Interval(ot.Point(inputSample.getDimension(), 0), ot.Point(inputSample.getDimension(), 1))

Comment (adumasphi) [reply lib/src/otmorris/Morris.hxx:45]: Yes we could do that but I am afraid that the user will forget to put the bounds when it is really needed.

Comment (sofianehaddad) [reply lib/src/otmorris/Morris.hxx:45]: Ok thus we should mention changes in Changelog

Comment (adumasphi) [reply lib/src/otmorris/Morris.hxx:45]: Yes right

**Comment by sofianehaddad** [comment]
hmm missing ChangeLog entry for API update

**Comment by jschueller** [comment]
I'll fix it.

File: ChangeLog [hunk 6]
```diff
@@ -1,3 +1,7 @@
+= 0.4 release (2018-03-20)
+
+ * bugfix: Take bounds into account in morris analysis
+
```
Comment (sofianehaddad) [inline ChangeLog:4]: please mention ticket number

Comment (jschueller) [reply ChangeLog:4]: done

**Comment by sofianehaddad** [comment]
Thanks

**Pull request merged**